File: Android-MVP-Java/app/src/main/java/com/base/mvp/core/di/component/ApplicationComponent.java
package com.base.mvp.core.di.component;

import android.app.Application;
import android.content.Context;

import com.base.mvp.MvpApplication;
import com.base.mvp.core.data.DataManager;
import com.base.mvp.core.di.ApplicationContext;
import com.base.mvp.core.di.module.ApplicationModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    void inject(MvpApplication app);

    @ApplicationContext
    Context context();

    Application application();

    DataManager getDataManager();
}

File: MVP-kotlin/app/src/main/java/com/base/mvp/core/di/components/ApplicationComponent.kt
package com.base.mvp.core.di.components

import android.app.Application
import android.content.Context
import com.base.mvp.MVPApplication
import com.base.mvp.core.di.ApplicationContext
import com.base.mvp.core.di.modules.ApplicationModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {
    fun inject(app: MVPApplication?)

    @ApplicationContext
    fun context(): Context?

    fun application(): Application?
}